api/desecapi/models.py

```diff
@@ -159,8 +160,16 @@ def send_email(self, reason, context=None, recipient=None):
         email.send()
 
 
+class TokenHashWithPlaintext(str):
```

Comment:
why attach the plain value on the hashed string rather then just attach it to the Token class?

Reply:
Having `plain` as a direct attribute of `Token` seems indeed more natural, but I somehow had gotten stuck on that route. I don't recall the exact reason, but I'll try to reproduce it and then we can see how to proceed.

Reply:
moved assignment of `self.plain` and `self.key` directly to `generate_key`